src/expr/src/expr/expr_split_part.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use risingwave_common::array::{
+    Array, ArrayBuilder, ArrayImpl, ArrayRef, DataChunk, Utf8ArrayBuilder,
+};
+use risingwave_common::error::{Result, RwError};
+use risingwave_common::types::DataType;
+use risingwave_common::{ensure, try_match_expand};
+use risingwave_pb::expr::expr_node::{RexNode, Type};
+use risingwave_pb::expr::ExprNode;
+
+use crate::expr::{build_from_prost, BoxedExpression, Expression};
+
+#[derive(Debug)]
+pub struct SplitPartExpression {
+    return_type: DataType,
+    string_expr: BoxedExpression,
+    delimiter_expr: BoxedExpression,
+    nth_expr: BoxedExpression,
+}
+
+impl Expression for SplitPartExpression {
+    fn return_type(&self) -> DataType {
+        self.return_type.clone()
+    }
+
+    fn eval(&self, input: &DataChunk) -> Result<ArrayRef> {
+        let string_columns = self.string_expr.eval(input)?;
+        let string_columns = string_columns.as_utf8();
+
+        let delimiter_columns = self.delimiter_expr.eval(input)?;
+        let delimiter_columns = delimiter_columns.as_utf8();
+
+        let nth_columns = self.nth_expr.eval(input)?;
+        let nth_columns = nth_columns.as_int64();
+
+        let row_len = input.cardinality();
+        let mut builder = Utf8ArrayBuilder::new(row_len)?;
+
+        for i in 0..row_len {
+            let string_column = string_columns.value_at(i);
+            let delimiter_column = delimiter_columns.value_at(i);
+            let nth_column = nth_columns.value_at(i);
+
+            // FIXME
```

Comment:
Looking for suggestions on:
- what error handling method to use in order to deal with the below three unwrapping
- what should be as expected behavior, when index is zero

Reply:
* The three unwrapping corresponding to `null`s at each position.
* As for expected behavior when an argument is `null` or when index is zero or when index out of bound, just try it in PostgreSQL.

Reply:
thx, fixed [dcc5b6a](https://github.com/singularity-data/risingwave/pull/2822/commits/dcc5b6a225a62d64fd5fc478bb0e741509e01899)